File: transferFile/WatchOS2TransferImage/WatchOS2TransferImage WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  WatchOS2TransferImage WatchKit Extension
//
//  Created by Thai, Kristina on 12/18/15.
//  Copyright © 2015 Thai, Kristina. All rights reserved.
//

#import "InterfaceController.h"
#import <WatchConnectivity/WatchConnectivity.h>

@interface InterfaceController() <WCSessionDelegate>
@property (weak, nonatomic) WCSession *session;
@end


@implementation InterfaceController

-(instancetype)init {
    self = [super init];
    
    if (self) {
        if ([WCSession isSupported]) {
            self.session = [WCSession defaultSession];
            self.session.delegate = self;
            [self.session activateSession];
        }
    }
    
    return self;
}

- (IBAction)sendDogs {
    if ([self.session isReachable]) {
        NSURL *filePath = [[NSURL alloc] initFileURLWithPath:[[NSBundle mainBundle] pathForResource:@"dogs" ofType:@"png"]];
        [self.session transferFile:filePath metadata:nil];
    }
}

-(void)session:(WCSession *)session didFinishFileTransfer:(WCSessionFileTransfer *)fileTransfer error:(NSError *)error {
    NSLog(@"error: %@", error);
}

@end

File: transferFile/WatchOS2TransferGif WatchKit Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  WatchOS2TransferImage WatchKit Extension
//
//  Created by Thai, Kristina on 11/8/15.
//  Copyright © 2015 Thai, Kristina. All rights reserved.
//

import WatchKit
import Foundation
import WatchConnectivity


class InterfaceController: WKInterfaceController, WCSessionDelegate {
    
    private let session : WCSession? = WCSession.isSupported() ? WCSession.defaultSession() : nil
    
    override init() {
        super.init()
        
        session?.delegate = self
        session?.activateSession()
    }
    
    @IBAction func sendDogs() {
        let filePath = NSURL.fileURLWithPath(NSBundle.mainBundle().pathForResource("dogs", ofType: "png")!)

        self.session?.transferFile(filePath, metadata: nil)

    }
    
    func session(session: WCSession, didFinishFileTransfer fileTransfer: WCSessionFileTransfer, error: NSError?) {
        print("error: ", error)
    }
}
